app/controllers/owners_controller.rb

```diff
@@ -53,4 +54,14 @@ def destroy
   def owner?
     @rubygem.owned_by?(current_user)
   end
+
+  def password_session_active?
+    session[:verification] && session[:verification] > Time.current
+  end
+
+  def render_verify_password
+    @redirect_uri = rubygem_owners_url(@rubygem)
+    render "passwords/verify"
+    false
```

Comment:
This seems hacky. Please create an action for this in the password controller. You can use params or session to pass redirect_uri.